docs/src/main/tut/chisel3/modules.md

```diff
@@ -84,3 +84,60 @@ children modules, using the ```Module``` constructor function and
 the Scala ```new``` keyword to create a
 new object.  We then wire them up to one another and to the ports of
 the ```Mux4``` interface.
+
+Note: Chisel `Module`s have an implicit clock (called `clock`) and
+an implicit reset (called `reset`). For different behavior, Chisel
+provides both `MultiIOModule` and `RawModule`.
+
+### `MultiIOModule`
+
+A `MultiIOModule` allows you to define as many different `IO` as needed
+and does not require you to implement an abstract member `io`.
+This can be useful when programmatically adding `IO` or adding `IO` via inheritance.
+An artifact of this is that Verilog generated from a `MultiIOModule` will
+*not* have the `io_` prefix. `MultiIOModule`s still have an implicit
+clock and reset like `Module`.
+
+<!-- TODO: Some example -->
+
+### `RawModule`
+
+`Module` are used to connect components inside a design. But once we
+want to synthesize for the real world (FPGA, ASIC, ...) some extra
+features are required like io specific naming, reset polarity, clock
+assignement, ...
```

Comment:
One other thing: can you remove this paragraph?

Reply:
ok